src/runtime_src/core/include/xclbin.h

```diff
@@ -38,6 +38,8 @@
 #ifndef _XCLBIN_H_
 #define _XCLBIN_H_
 
+#define MAX_DDR_BANKS    4
```

Comment:
I have just aligned the mem_topology declaration with Xocluser_INTF.h header file of XoclUser driver. There also, mem_topo _data declared with MAX_DDR_BANKS macro value.
https://github.com/Xilinx/MS_XOCLUSER/blob/master/Src/Inc/XoclUser_INTF.h#L257

If this macro usage is not recommended, then we need to invoke IOCTL_XOCL_STAT twice. 1st call is to get the mem_topo_data count, and 2nd call is to retrieve the mem_topology information.

Reply:
This macro is not desired.  4 would be too small anyway since mem tpoplogy includes more than just DDRs.  You need mem topology count from the data itself.  Either as you say, call ioctl twice, or over allocate a struct big enough to hold max and ioctl erroring out if preallocated size is too small.